src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java

```diff
@@ -23,20 +26,45 @@ public class DailyListProcessorImpl implements DailyListProcessor {
 
     private final DailyListRepository dailyListRepository;
     private final DailyListUpdater dailyListUpdater;
+    private final LogApi logApi;
 
     @Override
     public void processAllDailyLists() {
+
         stream(SourceType.values()).forEach(sourceType -> {
+
             var dailyListsGroupedByCourthouse = dailyListRepository.findByStatusAndStartDateAndSourceOrderByPublishedTimestampDesc(
                 JobStatusType.NEW,
                 now(),
                 String.valueOf(sourceType)
             ).stream().collect(groupingBy(DailyListEntity::getListingCourthouse));
 
-            dailyListsGroupedByCourthouse.forEach((listingCourthouse, dailyLists) -> processDailyListsForSourceType(dailyLists));
+            DailyListLogJobReport report = new DailyListLogJobReport(getCountOfDailyList(dailyListsGroupedByCourthouse), sourceType);
+            try {
+                dailyListsGroupedByCourthouse.forEach((listingCourthouse, dailyLists) -> {
+                    processDailyListsForSourceType(dailyLists, report);
+                });
+            } finally {
+                reportStats(report);
+            }
+
         });
     }
 
+    private int getCountOfDailyList(Map<String, List<DailyListEntity>> dailyListMap) {
+        int count = 0;
+        for (String entity : dailyListMap.keySet()) {
+            count = count + dailyListMap.get(entity).size();
+        }
+        return count;
+    }
+
+    private void reportStats(DailyListLogJobReport report) {
+        if (report.getTotalDailyListRegisteredJobStatus() > 0) {
+            logApi.processedDailyListJob(report);
```

Comment:
I think we should report the stats even if nothing was processed, that way we can see that a job was run.

Reply:
np. Done